tests/test_rust.py

```diff
@@ -287,3 +287,43 @@ def test_object_metric(tmp_path):
         assert "ip: Option<String>, " in content
         assert "trust: Option<String>, " in content
         assert "}" in content
+
+def test_dual_labeled_counter_metric(tmp_path):
+    """
+    Assert that an dual labeled counter metric is created.
+    """
+    translate.translate(
+        ROOT / "data" / "dual_labeled.yaml",
+        "rust",
+        tmp_path,
+        {"namespace": "Foo"},
+    )
+
+    assert set(x.name for x in tmp_path.iterdir()) == set(["glean_metrics.rs"])
+
+    with (tmp_path / "glean_metrics.rs").open("r", encoding="utf-8") as fd:
+        content = fd.read()
+        content = " ".join(content.split())
+
+        # TODO: work in progress, this is not yet fully implemented
+        assert "pub static dynamic_dynamic: ::glean::private::__export::Lazy<DualLabeledCounterMetric>" in content
+        assert "DualLabeledCounterMetric::new(meta, None, None)" in content
+        assert "pub static static_dynamic: ::glean::private::__export::Lazy<DualLabeledCounterMetric>" in content
+        assert 'DualLabeledCounterMetric::new(meta, Some(vec![::std::borrow::Cow::from("key1"), ::std::borrow::Cow::from("key2")]), None)' in content
+        assert "pub static dynamic_static: ::glean::private::__export::Lazy<DualLabeledCounterMetric>" in content
+        assert 'DualLabeledCounterMetric::new(meta, None, Some(vec![::std::borrow::Cow::from("category1"), ::std::borrow::Cow::from("category2")]))' in content
+        assert "pub static static_static: ::glean::private::__export::Lazy<DualLabeledCounterMetric>" in content
+        assert 'DualLabeledCounterMetric::new(meta, Some(vec![::std::borrow::Cow::from("key1"), ::std::borrow::Cow::from("key2")]), Some(vec![::std::borrow::Cow::from("category1"), ::std::borrow::Cow::from("category2")]))' in content
```

Comment:
I think more important tests here will be to assert that the yaml parses without exception or nit (so, `test_parser` or `test_metrics` instead, perhaps). The rust codegen, alas, is tested by the sample app in the Glean SDK (cf. https://bugzilla.mozilla.org/show_bug.cgi?id=1762858 ). 

We also need a negative test where bad-but-allowed-by-schema definitions are caught with appropriate errors.